fortio/cert.yaml

```diff
@@ -0,0 +1,119 @@
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: istio-system
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: ldemailly@google.com
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    # Enable the HTTP-01 challenge provider
+    http01: {}
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+  namespace: istio-system
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-v01.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: ldemailly@google.com
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    # Enable the HTTP-01 challenge provider
+    http01: {}
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: fortio-cert-prod
+  namespace: istio-system
+spec:
+  secretName: istio-ingress-certs
+  issuerRef:
+    name: letsencrypt-prod
+  commonName: fortio.istio.io
+  dnsNames:
+  - fortio.istio.io
+  - istio.fortio.org
+  acme:
+    config:
+    - http01:
+        # if we give it the right one it adds more specific rules that takeover
+        # ours and thus get 503 UF  because of auth but then we get a sync error
+        # but this one leaks: (TODO)
+        # E0112 03:02:54.090362       1 prepare.go:158] Error cleaning up solver: [istio.fortio.org] Error cleaning up ingress: error cleaning up ingress: Ingress.extensions "cert-manager-temp-ingress" is invalid: spec.rules[0].http.paths: Required value
+        # so "make cleanup" is needed after
+        ingress: cert-manager-temp-ingress
```

Comment:
@munnerz may I get your input here - is there a way to avoid that
(tell cert-manager to not try to change ingress rules - or alternatively to add `auth.istio.io/8089: NONE` to the svc it creates)

Reply:
@ldemailly if you set an `http01.ingressClass` instead of an `http01.ingress` then `cert-manager` will create its own temporary Ingress resource to handle the challenge, rather than modifying the existing Ingress. The temporary Ingress will be for the same domain name, and with the specified `ingress.class` annotation, but just for the `.well-known/acme/` path.

Reply:
Any ingress controller worth its salt will direct the traffic based on the domain+path to the backend service for the matching Ingress. If you think about it, this is necessary behavior, or you couldn't have two components of an application in different namespaces (since Ingresses are namespaced), e.g. `example.com/component-a` (in Namespace X) and `example.com/component-b` (in Namespace Y), you have to have two Ingresses for `example.com` (one for X one for Y) to combine those.

Istio ingress controller? Dunno. YMMV 😄

Reply:
@ldemailly just trying to gather some feedback on this feature request - do you need cert-manager to add some annotations to the *ingress* resource, or the *service*? (if this were to be added as a feature)

Reply:
As @whereisaaron states, `ingressClass` is intended to be used with ingress controllers that `flatten` ingresses (e.g. nginx). `ingress` is intended to be used by those that behave more like the GCE ingress controller (one ingress = one IP)

Reply:
@munnerz the service in this case (`auth.istio.io/8089: NONE` like https://github.com/istio/admin-sites/pull/32/files#diff-e3bb0a8ae5f27772213079b2b0ab44e6R67) but probably to the ingress too

Reply:
istio ingress does merge multiple ingress specs. I'm not super clear which value I should use still. will try though